src/fault_injector/fault_types/train_prio_fault.py

```diff
@@ -39,6 +40,11 @@ def resolve_fault(self, tick: int):
         :param tick: the simulation tick in which resolve_fault was called
         :type tick: Integer
         """
+        if self.train is None or self.train is not self.get_train(
+            self.simulation_object_updater, self.train.identifier
+        ):
+            raise ValueError("Train does not exist or fault not injected")
```

Comment:
But I do not actually think, that this is the best behavior. I think, that if the train despawned already, this should not raise an error. Rather, it should not change anything about the (now not existent) train and still log, that the fault has been resolved.